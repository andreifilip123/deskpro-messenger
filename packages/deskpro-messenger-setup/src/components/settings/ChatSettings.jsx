import React from 'react';
import PropTypes from 'prop-types';
import Immutable from 'immutable';
import {
  Drawer,
  Heading,
  Input,
  Label,
  Textarea,
  Select,
  Toggle
} from '@deskpro/react-components';

class ChatSettings extends React.PureComponent {
  static propTypes = {
    config: PropTypes.object,
    handleChange: PropTypes.func.isRequired,
    chatDepartments: PropTypes.object,
  };

  static defaultProps = {
    config: Immutable.fromJS({})
  };

  noAnswerOptions = [
    { value: '', label: 'Just show busy message' },
    { value: 'save_ticket', label: 'Save missed chat as ticket' },
    { value: 'create_ticket', label: 'Direct the user to new ticket' }
  ];

  handleSelectChange = (option, name) => {
    const value = typeof option === 'object' ? option.value : option;
    this.props.handleChange(value, name);
  };

  render() {
    const {
      config,
      handleChange,
      chatDepartments
    } = this.props;
    return (
      <Drawer>
        <Heading>Chat settings</Heading>
        <Toggle
          checked={config.getIn(['chat', 'enabled'])}
          name="chat.enabled"
          onChange={handleChange}
        >
          Enable chat
        </Toggle>
        <br />
        Prompt the user to describe their problem before the chat starts:
        <br />
        <Input
          type="text"
          value={config.getIn(['chat', 'prompt'])}
          name="chat.prompt"
          onChange={handleChange}
        />
        <h4>Who can use chat</h4>
        <h4>Pre-chat form</h4>
        <h4>Unanswered chat</h4>
        If no agents are online to accept a chat, or when the user has waited
        for{' '}
        <Input
          className="small"
          type="text"
          value={config.getIn(['chat', 'timeout'])}
          onChange={handleChange}
          name="chat.timeout"
        />{' '}
        seconds
        <br />
        <Select
          options={this.noAnswerOptions}
          value={config.getIn(['chat', 'noAnswerBehavior'], '')}
          onChange={this.handleSelectChange}
          name="chat.noAnswerBehavior"
        />
        <Label>Busy message</Label>
        <Textarea
          name="chat.busyMessage"
          id=""
          cols="40"
          rows="10"
          value={config.getIn(['chat', 'busyMessage'])}
          onChange={handleChange}
        />
        <br />
        <u>Missed chat ticket properties</u>
        <Label>Department</Label>
        <Select
          options={chatDepartments.toArray().map(dep => (
            {
              value: dep.get('id'),
              label: dep.get('title')
            }
          ))}
          value={config.getIn(['chat', 'ticketDefaults', 'department'])}
          onChange={this.handleSelectChange}
          name="chat.ticketDefaults.department"
        />
        <Label>Subject</Label>
        <Input
          type="text"
          value={config.getIn(['chat', 'ticketDefaults', 'subject'])}
          placeholder="Missed chat from {name}"
          onChange={handleChange}
          name="chat.ticketDefaults.subject"
        />
      </Drawer>
    );
  }
}
export default ChatSettings;
