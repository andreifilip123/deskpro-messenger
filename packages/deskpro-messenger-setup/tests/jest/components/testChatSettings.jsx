import React from 'react';
import renderer from 'react-test-renderer';
import ChatSettings from '../../../src/components/settings/ChatSettings';
import noop from '@deskpro/js-utils/dist/noop';
import Immutable from "immutable";

const mockMath = Object.create(global.Math);
mockMath.random = () => 0.5;
global.Math = mockMath;

const departments = [
  { id: 3, title: 'Sales' }, { id: 4, title: 'Support' }
];

it('+++capturing Snapshot of BooleanInput with true set', () => {
  const renderedValue = renderer.create(
    <ChatSettings
      handleChange={noop}
      chatDepartments={Immutable.fromJS(departments)}
    />
  ).toJSON();
  expect(renderedValue).toMatchSnapshot();
});
