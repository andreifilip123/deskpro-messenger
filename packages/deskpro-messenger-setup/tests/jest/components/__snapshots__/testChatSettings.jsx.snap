// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`+++capturing Snapshot of BooleanInput with true set 1`] = `
<li
  className="dp-list__element dp-column-drawer"
  data-dp-toggle-id={0}
>
  <h1
    className="dp-heading"
    onClick={[Function]}
  >
    Chat settings Â 
    <svg
      aria-hidden={true}
      className="svg-inline--fa fa-caret-down fa-w-10 fa-sm dp-icon dp-column-drawer__arrow"
      data-icon="caret-down"
      data-prefix="fas"
      onClick={[Function]}
      role="img"
      style={Object {}}
      viewBox="0 0 320 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M31.3 192h257.3c17.8 0 26.7 21.5 14.1 34.1L174.1 354.8c-7.8 7.8-20.5 7.8-28.3 0L17.2 226.1C4.6 213.5 13.5 192 31.3 192z"
        fill="currentColor"
        style={Object {}}
      />
    </svg>
  </h1>
  <section
    className="dp-section dp-section--hidden"
  >
    
    <div
      className="dp-input--toggle"
    >
      <input
        checked={false}
        className="right"
        id={null}
        name="chat.enabled"
        onChange={[Function]}
        type="checkbox"
        value=""
      />
      <label
        htmlFor={null}
        onClick={[Function]}
      >
        Enable chat
      </label>
    </div>
    <br />
    Prompt the user to describe their problem before the chat starts:
    <br />
    <div
      className="dp-input"
    >
      <input
        id="id1"
        name="chat.prompt"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        type="text"
        value={undefined}
      />
    </div>
    <label
      className="dp-input__label"
      htmlFor=""
    >
      Default chat department
      
    </label>
    <div
      className="dp-select"
    >
      <div
        className="react-select css-2b097c-container"
        id={undefined}
        onKeyDown={[Function]}
      >
        <div
          className="react-select__control css-yk16xz-control"
          onMouseDown={[Function]}
          onTouchEnd={[Function]}
        >
          <div
            className="react-select__value-container css-1hwfws3"
          >
            <div
              className="react-select__placeholder css-1wa3eu0-placeholder"
            >
              Please select
            </div>
            <div
              className="css-w8afj7-Input"
            >
              <div
                className="react-select__input"
                style={
                  Object {
                    "display": "inline-block",
                  }
                }
              >
                <input
                  aria-autocomplete="list"
                  aria-label={undefined}
                  aria-labelledby={undefined}
                  autoCapitalize="none"
                  autoComplete="off"
                  autoCorrect="off"
                  className={undefined}
                  disabled={false}
                  id="react-select-2-input"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  spellCheck="false"
                  style={
                    Object {
                      "background": 0,
                      "border": 0,
                      "boxSizing": "content-box",
                      "color": "inherit",
                      "fontSize": "inherit",
                      "label": "input",
                      "opacity": 1,
                      "outline": 0,
                      "padding": 0,
                      "width": "1px",
                    }
                  }
                  tabIndex="0"
                  type="text"
                  value=""
                />
                <div
                  style={
                    Object {
                      "height": 0,
                      "left": 0,
                      "overflow": "scroll",
                      "position": "absolute",
                      "top": 0,
                      "visibility": "hidden",
                      "whiteSpace": "pre",
                    }
                  }
                >
                  
                </div>
              </div>
            </div>
          </div>
          <div
            className="react-select__indicators css-1g48xl4-IndicatorsContainer"
          >
            <span
              className="react-select__indicator-separator css-1okebmr-indicatorSeparator"
            />
            <div
              className="react-select__dropdown-indicator"
            />
          </div>
        </div>
        <input
          name="chat.department"
          type="hidden"
          value=""
        />
      </div>
    </div>
    <h4>
      Who can use chat
    </h4>
    <h4>
      Pre-chat form
    </h4>
    <h4>
      Unanswered chat
    </h4>
    If no agents are online to accept a chat, or when the user has waited for
     
    <div
      className="dp-input small"
    >
      <input
        id="id2"
        name="chat.timeout"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        type="text"
        value={undefined}
      />
    </div>
     
    seconds
    <br />
    <div
      className="dp-select"
    >
      <div
        className="react-select css-2b097c-container"
        id={undefined}
        onKeyDown={[Function]}
      >
        <div
          className="react-select__control css-yk16xz-control"
          onMouseDown={[Function]}
          onTouchEnd={[Function]}
        >
          <div
            className="react-select__value-container react-select__value-container--has-value css-1hwfws3"
          >
            <div
              className="react-select__single-value css-1uccc91-singleValue"
            >
              Just show busy message
            </div>
            <div
              className="css-w8afj7-Input"
            >
              <div
                className="react-select__input"
                style={
                  Object {
                    "display": "inline-block",
                  }
                }
              >
                <input
                  aria-autocomplete="list"
                  aria-label={undefined}
                  aria-labelledby={undefined}
                  autoCapitalize="none"
                  autoComplete="off"
                  autoCorrect="off"
                  className={undefined}
                  disabled={false}
                  id="react-select-3-input"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  spellCheck="false"
                  style={
                    Object {
                      "background": 0,
                      "border": 0,
                      "boxSizing": "content-box",
                      "color": "inherit",
                      "fontSize": "inherit",
                      "label": "input",
                      "opacity": 1,
                      "outline": 0,
                      "padding": 0,
                      "width": "1px",
                    }
                  }
                  tabIndex="0"
                  type="text"
                  value=""
                />
                <div
                  style={
                    Object {
                      "height": 0,
                      "left": 0,
                      "overflow": "scroll",
                      "position": "absolute",
                      "top": 0,
                      "visibility": "hidden",
                      "whiteSpace": "pre",
                    }
                  }
                >
                  
                </div>
              </div>
            </div>
          </div>
          <div
            className="react-select__indicators css-1g48xl4-IndicatorsContainer"
          >
            <span
              className="react-select__indicator-separator css-1okebmr-indicatorSeparator"
            />
            <div
              className="react-select__dropdown-indicator"
            />
          </div>
        </div>
        <input
          name="chat.noAnswerBehavior"
          type="hidden"
          value=""
        />
      </div>
    </div>
    <label
      className="dp-input__label"
      htmlFor=""
    >
      Busy message
      
    </label>
    <div
      className="dp-textarea"
    >
      <textarea
        cols="40"
        id=""
        name="chat.busyMessage"
        onChange={[Function]}
        rows="10"
        value=""
      />
    </div>
    <br />
    <u>
      Missed chat ticket properties
    </u>
    <label
      className="dp-input__label"
      htmlFor=""
    >
      Department
      
    </label>
    <div
      className="dp-select"
    >
      <div
        className="react-select css-2b097c-container"
        id={undefined}
        onKeyDown={[Function]}
      >
        <div
          className="react-select__control css-yk16xz-control"
          onMouseDown={[Function]}
          onTouchEnd={[Function]}
        >
          <div
            className="react-select__value-container css-1hwfws3"
          >
            <div
              className="react-select__placeholder css-1wa3eu0-placeholder"
            >
              Please select
            </div>
            <div
              className="css-w8afj7-Input"
            >
              <div
                className="react-select__input"
                style={
                  Object {
                    "display": "inline-block",
                  }
                }
              >
                <input
                  aria-autocomplete="list"
                  aria-label={undefined}
                  aria-labelledby={undefined}
                  autoCapitalize="none"
                  autoComplete="off"
                  autoCorrect="off"
                  className={undefined}
                  disabled={false}
                  id="react-select-4-input"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  spellCheck="false"
                  style={
                    Object {
                      "background": 0,
                      "border": 0,
                      "boxSizing": "content-box",
                      "color": "inherit",
                      "fontSize": "inherit",
                      "label": "input",
                      "opacity": 1,
                      "outline": 0,
                      "padding": 0,
                      "width": "1px",
                    }
                  }
                  tabIndex="0"
                  type="text"
                  value=""
                />
                <div
                  style={
                    Object {
                      "height": 0,
                      "left": 0,
                      "overflow": "scroll",
                      "position": "absolute",
                      "top": 0,
                      "visibility": "hidden",
                      "whiteSpace": "pre",
                    }
                  }
                >
                  
                </div>
              </div>
            </div>
          </div>
          <div
            className="react-select__indicators css-1g48xl4-IndicatorsContainer"
          >
            <span
              className="react-select__indicator-separator css-1okebmr-indicatorSeparator"
            />
            <div
              className="react-select__dropdown-indicator"
            />
          </div>
        </div>
        <input
          name="chat.ticketDefaults.department"
          type="hidden"
          value=""
        />
      </div>
    </div>
    <label
      className="dp-input__label"
      htmlFor=""
    >
      Subject
      
    </label>
    <div
      className="dp-input"
    >
      <input
        id="id3"
        name="chat.ticketDefaults.subject"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        placeholder="Missed chat from {name}"
        type="text"
        value={undefined}
      />
    </div>
  </section>
</li>
`;
